bot/exts/info/information.py

```diff
@@ -518,15 +520,26 @@ def add_content(title: str, content: str) -> None:
                 title = f"Raw {field_name} ({current}/{total})"
                 add_content(title, transformer(item))
 
-        for page in paginator.pages:
-            await ctx.send(page, allowed_mentions=AllowedMentions.none())
+        return paginator
 
     @cooldown_with_role_bypass(2, 60 * 3, BucketType.member, bypass_roles=constants.STAFF_PARTNERS_COMMUNITY_ROLES)
     @group(invoke_without_command=True)
     @in_whitelist(channels=(constants.Channels.bot_commands,), roles=constants.STAFF_PARTNERS_COMMUNITY_ROLES)
     async def raw(self, ctx: Context, message: Message) -> None:
         """Shows information about the raw API response."""
-        await self.send_raw_content(ctx, message)
+        if not message.channel.permissions_for(ctx.author).read_messages:
+            await ctx.send(":x: You do not have permissions to see the channel this message is in.")
+            return
+
+        paginator = await self.get_raw_content(message)
+
+        for page in paginator.pages:
+            await ctx.send(page, allowed_mentions=AllowedMentions.none())
+
+    async def _raw_context_menu_callback(self, interaction: Interaction, message: Message) -> None:
+        """The callback to be ran when the Raw Text context menu button is used."""
+        paginator = await self.get_raw_content(message)
+        await interaction.response.send_message(paginator.pages[1], ephemeral=True)
```

Comment:
Perhaps we could trail out some button-based pagination here? It's a shame that users can only see one page with the context menu version. I think one of the largest (pun intended) issues people had with buttons for pagination is their large vertical size, though it may be less of an issue in ephemeral messages.

Reply:
Yeah, okay. I'll try adding a view with some buttons to see if we can scroll through.

Reply:
An alternative would be to do what was suggested in https://github.com/python-discord/bot/issues/1037 and just send a pastebin link if the message is too long, maybe in a similar way to the eval command where it displays as much as it can first.